packages/shared_preferences/shared_preferences/lib/shared_preferences.dart

```diff
@@ -146,6 +180,19 @@ class SharedPreferences {
   /// Completes with true once the user preferences for the app has been cleared.
   Future<bool> clear() {
     _preferenceCache.clear();
+    if (_prefixHasBeenChanged) {
+      try {
+        return _store.clearWithPrefix(_prefix);
+      } catch (e) {
+        // Catching and clarifying UnimplementedError to provide a more robust message.
+        if (e is UnimplementedError) {
+          throw UnimplementedError('''
+This implementation of Shared Preferences doesn't yet support the setPrefix method.
+Either update the implementation to support setPrefix, or do not call setPrefix.
+        ''');
+        }
```

Comment:
`else { rethrow; }`

Otherwise you're silently eating all other exceptions/errors.